lib/SheetClip/SheetClip.js

```diff
@@ -9,59 +9,85 @@
  * Licensed under the MIT license.
  * http://github.com/warpech/sheetclip/
  */
-/*jslint white: true*/
-(function (global) {
-  "use strict";
+(((global) => {
+  const regUniversalNewLine = /^(\r\n|\n\r|\r|\n)/;
+  const regNextCellNoQuotes = /^[^\t\r\n]+/;
+  const regNextEmptyCell = /^\t/;
 
-  function countQuotes(str) {
-    return str.split('"').length - 1;
-  }
-
-  var SheetClip = {
+  const SheetClip = {
     /**
      * Decode spreadsheet string into array
      *
      * @param {String} str
      * @returns {Array}
      */
-    parse: function (str) {
-      var r, rLen, rows, arr = [], a = 0, c, cLen, multiline, last;
+    parse(str) {
+      const arr = [['']];
 
-      rows = str.replace(/\r\n|\r/g, '\n').split('\n');
-
-      if (rows.length > 1 && rows[rows.length - 1] === '') {
-        rows.pop();
+      if (str.length === 0) {
+        return arr;
       }
-      for (r = 0, rLen = rows.length; r < rLen; r += 1) {
-        rows[r] = rows[r].split('\t');
 
-        for (c = 0, cLen = rows[r].length; c < cLen; c += 1) {
-          if (!arr[a]) {
-            arr[a] = [];
-          }
-          if (multiline && c === 0) {
-            last = arr[a].length - 1;
-            arr[a][last] = arr[a][last] + '\n' + rows[r][0];
+      let column = 0;
+      let row = 0;
+      let lastLength;
 
-            if (multiline && (countQuotes(rows[r][0]) & 1)) { //& 1 is a bitwise way of performing mod 2
-              multiline = false;
-              arr[a][last] = arr[a][last].substring(0, arr[a][last].length - 1).replace(/""/g, '"');
-            }
-          }
-          else {
-            if (c === cLen - 1 && rows[r][c].indexOf('"') === 0 && (countQuotes(rows[r][c]) & 1)) {
-              arr[a].push(rows[r][c].substring(1).replace(/""/g, '"'));
-              multiline = true;
-            }
-            else {
-              arr[a].push(rows[r][c].replace(/""/g, '"'));
-              multiline = false;
+      while (str.length > 0) {
+        if (lastLength === str.length) {
+          // In the case If in last cycle we didn't match anything, we have to leave the infinite loop
+          break;
+        }
```

Comment:
Can we somehow combine this into one condition? 

`while (str.length > 0 && lastLength !== str.length)`

Or isn't there a better loop type for this use case?

Reply:
As we discussed at a meeting, we can merge it without this suggestion.